### PR TITLE
adding plop keyboard

### PR DESCRIPTION
**Pull request by lukeski14**
## Description

Adding plop keyboard
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rainkeebs/plop/plop.c [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Copyright 2023 Regan Palmer (@rainkeebs)
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/plop.c:1]: This file should be removed. (Assuming the matrix data in `info.json` is correct)

File: keyboards/rainkeebs/plop/plop.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 Regan Palmer (@rainkeebs)
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/plop.h:1]: This file should be removed. (Assuming the matrix data in `info.json` is correct)

File: keyboards/rainkeebs/plop/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# plop
+
+![plop](https://imgur.com/5zZRGun)
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/readme.md:3]:
```suggestion
![plop](https://i.imgur.com/5zZRGunh.png)
```

File: keyboards/rainkeebs/plop/config.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/config.h:40]:
```suggestion
```

File: keyboards/rainkeebs/plop/info.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    "maintainer": "rainkeebs",
+    "bootloader": "atmel-dfu",
+    "processor": "atmega32u4",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x7070",
+        "vid": "0x726B"
+    },
+    "layouts": {
+        "LAYOUT_ortho_4x12": {
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/info.json:13]:
```suggestion
        "LAYOUT": {
```
\+ whatever other fixes this entails.

File: keyboards/rainkeebs/plop/rules.mk [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/rules.mk:4]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by lukeski14** [comment]
Let me know if additional changes are needed @waffle87 ! Appreciate it

File: keyboards/rainkeebs/plop/rules.mk [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/rainkeebs/plop/rules.mk:10]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/rainkeebs/plop/rules.mk [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/rainkeebs/plop/rules.mk:4]:
```suggestion
BOOTMAGIC_ENABLE = yes	# Enable bootmagic Lite
```

File: keyboards/rainkeebs/plop/rules.mk [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (drashna) [inline keyboards/rainkeebs/plop/rules.mk:13]:
```suggestion
RGBLIGHT_ENABLE = yes

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by lukeski14** [comment]
I believe this one is good to go still. Let me know, thanks @drashna @waffle87

File: keyboards/rainkeebs/plop/config.h [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 4
+#define MATRIX_COLS 12
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/config.h:26]:
```suggestion
```

File: keyboards/rainkeebs/plop/config.h [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, D5, B3, B2 }
+#define MATRIX_COL_PINS { C7, C6, B6, B5, B4, D7, D6, D4, F6, F7, F5, F4 }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING 5
+
```
Comment (waffle87) [inline keyboards/rainkeebs/plop/config.h:37]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/rainkeebs/plop/config.h [hunk 12]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 Regan Palmer (@rainkeebs)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, D5, B3, B2 }
+#define MATRIX_COL_PINS { C7, C6, B6, B5, B4, D7, D6, D4, F6, F7, F5, F4 }
```
Comment (drashna) [inline keyboards/rainkeebs/plop/config.h:8]: This should be moved to the info.json

**Pull request closed without merging**